### PR TITLE
Market detail UI

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart [hunk 1]
```diff
@@ -11,33 +11,39 @@ class DashedDivider extends StatelessWidget {
...
+                return Padding(
+                  padding: EdgeInsets.only(
+                    top: topPadding,
+                  ),
+                  child: SizedBox(
+                    width: dashWidth,
+                    height: 1.h,
+                    child: const DecoratedBox(
+                      decoration: BoxDecoration(
+                        color: Color(0xFFE0E5EB),
```
Comment (eli1stark) [inline plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart:37]: ??

Comment (VovaMedvediev) [reply plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart]: done

File: lib/app/screens/news/news.dart [hunk 2]
```diff
@@ -75,6 +76,10 @@ class _NewsState extends State<News> {
                         context,
                         news.news[index].urlAddress,
                       ),
+                      height: 110.h,
```
Comment (eli1stark) [inline lib/app/screens/news/news.dart:79]: remove

File: lib/app/screens/news/news.dart [hunk 3]
```diff
@@ -75,6 +76,10 @@ class _NewsState extends State<News> {
...
+                      padding: EdgeInsets.symmetric(
+                        vertical: 20.h,
```
Comment (eli1stark) [inline lib/app/screens/news/news.dart:81]: remove

File: lib/app/shared/features/chart/view/asset_chart.dart [hunk 4]
```diff
@@ -38,17 +38,27 @@ class AssetChart extends HookWidget {
             candleResolution: chartState.resolution,
             candles: chartState.candles,
             onCandleSelected: onCandleSelected,
-            selectedCandlePadding: 15.w,
+            chartHeight: 240.h,
```
Comment (eli1stark) [inline lib/app/shared/features/chart/view/asset_chart.dart:41]: remove

File: lib/app/shared/features/chart/view/asset_chart.dart [hunk 5]
```diff
@@ -38,17 +38,27 @@ class AssetChart extends HookWidget {
...
+            chartWidgetHeight: 336.h,
```
Comment (eli1stark) [inline lib/app/shared/features/chart/view/asset_chart.dart:42]: remove

File: lib/app/shared/features/chart/view/asset_chart.dart [hunk 6]
```diff
@@ -38,17 +38,27 @@ class AssetChart extends HookWidget {
...
+            isAssetChart: true,
+          ),
+          loading: () => LoadingChartView(
+            chartHeight: 240.h,
```
Comment (eli1stark) [inline lib/app/shared/features/chart/view/asset_chart.dart:46]: remove

File: lib/app/shared/features/chart/view/asset_chart.dart [hunk 7]
```diff
@@ -38,17 +38,27 @@ class AssetChart extends HookWidget {
...
+            chartWidgetHeight: 336.h,
```
Comment (eli1stark) [inline lib/app/shared/features/chart/view/asset_chart.dart:47]: remove

File: lib/app/shared/features/chart/view/asset_chart.dart [hunk 8]
```diff
@@ -38,17 +38,27 @@ class AssetChart extends HookWidget {
...
           error: (String error) {
             return Center(
               child: Text(error),
             );
           },
         );
       },
-      loading: () => const LoadingChartView(),
+      loading: () => LoadingChartView(
+        chartHeight: 240.h,
```
Comment (eli1stark) [inline lib/app/shared/features/chart/view/asset_chart.dart:58]: remove

File: lib/app/shared/features/chart/view/asset_chart.dart [hunk 9]
```diff
@@ -38,17 +38,27 @@ class AssetChart extends HookWidget {
...
+        chartWidgetHeight: 336.h,
```
Comment (eli1stark) [inline lib/app/shared/features/chart/view/asset_chart.dart:59]: remove

File: lib/app/shared/features/market_details/view/components/about_block/components/clickable_underlined_text.dart [hunk 10]
```diff
@@ -13,13 +14,22 @@ class ClickableUnderlinedText extends StatelessWidget {
...
-        text: text,
-        recognizer: TapGestureRecognizer()..onTap = onTap,
-        style: const TextStyle(
-          color: Colors.black54,
-          decoration: TextDecoration.underline,
+    return InkWell(
+      onTap: onTap,
+      child: Container(
+        padding: EdgeInsets.only(
+          bottom: 3.h,
```
Comment (eli1stark) [inline lib/app/shared/features/market_details/view/components/about_block/components/clickable_underlined_text.dart:21]: remove

File: lib/app/shared/features/market_details/view/components/about_block/components/clickable_underlined_text.dart [hunk 11]
```diff
@@ -13,13 +14,22 @@ class ClickableUnderlinedText extends StatelessWidget {
...
+        ),
+        decoration: BoxDecoration(
+          border: Border(
+            bottom: BorderSide(
+              width: 2.h,
```
Comment (eli1stark) [inline lib/app/shared/features/market_details/view/components/about_block/components/clickable_underlined_text.dart:26]: remove

File: lib/app/shared/features/market_details/view/components/market_news_block/market_news_block.dart [hunk 12]
```diff
@@ -29,31 +31,48 @@ class MarketNewsBlock extends HookWidget {
...
+                  ),
+                  timestamp: formatNewsDate(
+                    news.news[index].timestamp,
+                  ),
+                  onTap: () => launchURL(
+                    context,
+                    news.news[index].urlAddress,
+                  ),
+                  padding: EdgeInsets.only(
+                    bottom: 24.h,
```
Comment (eli1stark) [inline lib/app/shared/features/market_details/view/components/market_news_block/market_news_block.dart:60]: remove

File: lib/app/shared/features/market_details/view/market_details.dart [hunk 13]
```diff
@@ -33,58 +34,81 @@ class MarketDetails extends HookWidget {
...
-        appBar: MarketDetailsAppBar(
-          marketItem: marketItem,
+    return SPageFrame(
+      header: SPaddingH24(
+        child: SSmallHeader(
+          title: '${marketItem.name} (${marketItem.id})',
+          showStarButton: true,
+          isStarSelected:
+              watchlistIdsN.isInWatchlist(marketItem.associateAsset),
+          onBackButtonTap: () => Navigator.of(context).pop(),
```
Comment (eli1stark) [inline lib/app/shared/features/market_details/view/market_details.dart:47]: redundant argument. Headers by default use pop if parameter is null

File: plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart [hunk 14]
```diff
@@ -11,33 +13,39 @@ class DashedDivider extends StatelessWidget {
...
-                padding: EdgeInsets.only(top: topPadding),
-                child: SizedBox(
-                  width: dashWidth,
-                  height: 1.h,
-                  child: const DecoratedBox(
-                    decoration: BoxDecoration(
-                      color: Color(0xFFE0E5EB),
+    return Container(
+      width: 1.sw,
+      padding: EdgeInsets.only(left: 24.w),
```
Comment (eli1stark) [inline plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart:18]: remove

File: plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart [hunk 15]
```diff
@@ -11,33 +13,39 @@ class DashedDivider extends StatelessWidget {
...
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          final boxWidth = constraints.constrainWidth();
+          final dashWidth = 5.w;
```
Comment (eli1stark) [inline plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart:22]: remove

File: plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart [hunk 16]
```diff
@@ -11,33 +13,39 @@ class DashedDivider extends StatelessWidget {
...
+            children: List.generate(
+              dashCount,
+              (_) {
+                return Padding(
+                  padding: EdgeInsets.only(
+                    top: topPadding,
+                  ),
+                  child: SizedBox(
+                    width: dashWidth,
+                    height: 1.h,
```
Comment (eli1stark) [inline plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart:36]: remove

File: plugins/simple_chart/lib/renderer/chart_painter.dart [hunk 17]
```diff
@@ -263,9 +265,9 @@ class ChartPainter extends BaseChartPainter {
 
   @override
   void drawMaxAndMin(Canvas canvas) {
-    if (!isLongPress) {
+    if (!isLongPress && isAssetChart) {
       //Plot the maximum and minimum values
-      const x = 0.0;
+      final x = 24.w;
```
Comment (eli1stark) [inline plugins/simple_chart/lib/renderer/chart_painter.dart:270]: remove

File: plugins/simple_chart/lib/renderer/chart_painter.dart [hunk 18]
```diff
@@ -263,9 +265,9 @@ class ChartPainter extends BaseChartPainter {
...
       var y = getMainY(mMainLowMinValue!) + 20.h;
```
Comment (eli1stark) [inline plugins/simple_chart/lib/renderer/chart_painter.dart:271]: remove

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart]: done

File: lib/app/screens/news/news.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/app/screens/news/news.dart:79]: done

File: lib/app/screens/news/news.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/app/screens/news/news.dart:81]: done

File: lib/app/shared/features/chart/view/asset_chart.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/chart/view/asset_chart.dart:41]: done

File: lib/app/shared/features/chart/view/asset_chart.dart (hunk 5 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/chart/view/asset_chart.dart:42]: +

Comment (VovaMedvediev) [reply lib/app/shared/features/chart/view/asset_chart.dart:42]: +

Comment (VovaMedvediev) [reply lib/app/shared/features/chart/view/asset_chart.dart:42]: done

File: lib/app/shared/features/chart/view/asset_chart.dart (hunk 6 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/chart/view/asset_chart.dart:46]: done

File: lib/app/shared/features/chart/view/asset_chart.dart (hunk 7 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/chart/view/asset_chart.dart:47]: done

File: lib/app/shared/features/chart/view/asset_chart.dart (hunk 8 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/chart/view/asset_chart.dart:58]: done

File: lib/app/shared/features/chart/view/asset_chart.dart (hunk 9 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/chart/view/asset_chart.dart:59]: done

File: lib/app/shared/features/market_details/view/components/about_block/components/clickable_underlined_text.dart (hunk 10 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/market_details/view/components/about_block/components/clickable_underlined_text.dart:21]: done

File: lib/app/shared/features/market_details/view/components/about_block/components/clickable_underlined_text.dart (hunk 11 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/market_details/view/components/about_block/components/clickable_underlined_text.dart:26]: done

File: lib/app/shared/features/market_details/view/components/market_news_block/market_news_block.dart (hunk 12 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/market_details/view/components/market_news_block/market_news_block.dart:60]: done

File: lib/app/shared/features/market_details/view/market_details.dart (hunk 13 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/market_details/view/market_details.dart:47]: done

File: plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart (hunk 14 above)
Comment (VovaMedvediev) [reply plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart:18]: done

File: plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart (hunk 15 above)
Comment (VovaMedvediev) [reply plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart:22]: done

File: plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart (hunk 16 above)
Comment (VovaMedvediev) [reply plugins/simple_chart/lib/components/dashed_divider/dashed_divider.dart:36]: done

File: plugins/simple_chart/lib/renderer/chart_painter.dart (hunk 17 above)
Comment (VovaMedvediev) [reply plugins/simple_chart/lib/renderer/chart_painter.dart:270]: done

File: plugins/simple_chart/lib/renderer/chart_painter.dart (hunk 18 above)
Comment (VovaMedvediev) [reply plugins/simple_chart/lib/renderer/chart_painter.dart:271]: done

**Pull request merged**